py/selenium/webdriver/support/wait.py

```diff
@@ -92,7 +93,9 @@ def __init__(
     def __repr__(self) -> str:
         return f'<{type(self).__module__}.{type(self).__name__} (session="{self._driver.session_id}")>'
 
-    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = "") -> T:
+    def until(
+        self, method: Callable[[D], Union[Literal[False], T]], message: Union[str, Callable[[Any], str]] = ""
```

Comment:
Thank you for your work.
I think it would be better to indicate that the Callable cannot take any arguments.

```suggestion
        self, method: Callable[[D], Union[Literal[False], T]], message: Union[str, Callable[[], str]] = ""
```

Reply:
Will this work for scenarios like `lambda x: f'Message: {x.current_url}'`?

Reply:
The current implementation is `final_msg = message() if callable(message) else message`, so `lambda x: f'Message: {x.current_url}'` will result in an error because `x` is None. Instead, the following scenario will occur:

```python
WebDriverWait(driver, 10).until(lambda x: x.find_element(By.ID, "someId"), lambda: f'Message: {driver.current_url}')
```

Reply:
Fair, thanks for pointing this out :smile: